esp-hal/src/spi/slave.rs

```diff
@@ -140,7 +141,10 @@ impl<'d> Spi<'d, Blocking> {
     #[instability::unstable]
     pub fn with_miso(self, miso: impl PeripheralOutput<'d>) -> Self {
         let miso = miso.into();
-        miso.set_to_push_pull_output();
+
+        miso.apply_output_config(&OutputConfig::default());
+        miso.set_output_enable(true);
+
         self.spi.info().miso.connect_to(&miso);
```

Comment:
I can't help but feel that these should/could be a one liner. Maybe a helper function type deal.

Reply:
Yeah similar like `with_data_pin` in the lcd driver, maybe.